azuredevops/internal/service/core/data_project.go

```diff
@@ -17,8 +19,22 @@ func DataProject() *schema.Resource {
 				Type:     v.Type,
 				Computed: true,
 			}
+		} else {
+			v.Required = false
+			v.Optional = true
+			v.ConflictsWith = []string{
+				"project_id",
```

Comment:
`GetProject` API doesn't care what's the param we send,  `projectName` or `projectID`,  a project will return if exist. So I think users should configure at least one of them or both.  We have two optional here:
1. Remove the `ConflictsWith`, users can configure both of them
2. Add `ConflictsWith` for `project_name` too, only one is needed here

Personal prefer optional 2

Reply:
if you have conflicts on one, you already can't have both of them, so the second conflict would be redundant. I've added a test to show this :)